### PR TITLE
Removed from models import* #fixed 1057

### PR DESCRIPTION
**Pull request by ankit01ojha**
Replaced `bodhi.server.models` package to `bodhi.server.models` module.

### CODE REVIEW

**Comment by ankit01ojha** [comment]
@bowlofeggs can you please review this :)

**Comment by trishnaguha** [comment]
@ankit01ojha Looks like you already opened several PRs for this issue before. When modifying commit it is recommended to do force push instead of closing the PR and opening a new one.

**Comment by ankit01ojha** [comment]
@trishnaguha  I am really sorry for this, I will not do it again( I was new to github, now I am quite used to it).

File: bodhi/server/models.py [hunk 1]
```diff
@@ -34,7 +34,7 @@
 
 
 from sqlalchemy import Unicode, UnicodeText, Integer, Boolean
-from sqlalchemy import DateTime, engine_from_config
+from sqlalchemy import DateTime
```
Comment (bowlofeggs) [inline bodhi/server/models.py]: We shouldn't be making very many modifications to this file, other than merging the Enum items in. This import is needed, and I see a lot of other unrelated changes elsewhere in the file.

**Review by bowlofeggs — CHANGES_REQUESTED** [review]
We don't want to make changes that are unrelated to renaming the module. Also, the tests do not pass.

**Comment by ankit01ojha** [comment]
@bowlofeggs Modified!

File: bodhi/server/models.py [hunk 2]
```diff
@@ -158,12 +266,10 @@ def grid_columns(cls):
 
     def update_relationship(self, name, model, data, db):
         """Add items to or remove items from a many-to-many relationship
-
```
Comment (bowlofeggs) [inline bodhi/server/models.py]: Dropping all of these spaces are not related to the change - can you make sure that only the ```git mv``` and merging enum in are happening in this pull request?

**Comment by ankit01ojha** [comment]
@bowlofeggs Is this fine now or still needs some improvement?

**Comment by ankit01ojha** [comment]
@bowlofeggs Is this fine or it still needs corrections?

**Comment by trishnaguha** [comment]
@ankit01ojha Can you run the testsuite with these changes and update the ticket with the result of your test (Result as in Success or Failing). That's the best way which lets you know if you are doing it right when dealing with large codebase. 
To run the testsuite follow the Doc: https://github.com/fedora-infra/bodhi/blob/develop/docs/developer_docs.rst#quick-tips-about-the-bodhi-vagrant-environment

**Comment by ankit01ojha** [comment]
@trishnaguha I tried to do this https://github.com/fedora-infra/bodhi/blob/develop/docs/developer_docs.rst#quick-tips-about-the-bodhi-vagrant-environment as you said but I can't find anything in my in .bashrc so ran the `$ nosetests -v` here's the output all tests passed .

screenshot from 2017-02-05 11-30-54

**Comment by ankit01ojha** [comment]
@trishnaguha  I tried to do this https://github.com/fedora-infra/bodhi/blob/develop/docs/developer_docs.rst#quick-tips-about-the-bodhi-vagrant-environment as you said but I can't find set of commands in my `.bashrc` so I ran this `$ nosetests -v`  and here's the output.

screenshot from 2017-02-05 11-42-52

**Comment by ankit01ojha** [comment]
I am confused does it need changes, as error shown are the files that I have removed.

**Comment by bowlofeggs** [comment]
I am guessing that the old ```.pyc``` file ```__init__.pyc``` may still be present on your system.

**Comment by ankit01ojha** [comment]
@bowlofeggs this time it passed 79% other errors are coming due to the import of `bodhi.server.models` and its files, so, do I have to make changes in all other file in this commit or I make other commit for them?
screenshot from 2017-02-07 13-53-39

**Comment by ankit01ojha** [comment]
@bowlofeggs After removing the old `.pyc` files and making some corrections I ran the test and it passed 79%, other errors are due to the other files which are importing models.
screenshot from 2017-02-07 13-53-39

**Comment by trishnaguha** [comment]
@ankit01ojha Can you please paste the failed tests in `fpaste`?

**Comment by ankit01ojha** [comment]
@trishnaguha Here's it is  https://da.gd/1ccA

**Comment by bowlofeggs** [comment]
@ankit01ojha The issue is that there are many places in the code that import the file that has been deleted, so you will need to adjust the imports so they import the new location. I recommend going through a Python tutorial or two to learn more about modules and import statements. The official Python 2 tutorial is decent, and it has a page about modules:

https://docs.python.org/2/tutorial/index.html
https://docs.python.org/2/tutorial/modules.html

**Comment by ankit01ojha** [comment]
@bowlofeggs I almost fixed all the import errors, now its 81% passed which is the minimum required.
The rest errors are https://da.gd/CeOkT. Tried to fix them but was not successful.
screenshot from 2017-02-12 00-07-04

**Comment by trishnaguha** [comment]
@ankit01ojha You need to follow the documentation that bowlofeggs suggested and fix import path. If you can look in to the failed testcases in the fpaste, it gives you hints exactly where your code is failing.

**Comment by bowlofeggs** [comment]
@ankit01ojha It looks like there are mock statements that are altering the module you've removed. You should be able to look in the cited test files and find those paths. They're often going to be in statements that look like ```@patch('bodhi.server.models.models.') over the tests, but they can also appear inside of tests or start with ```@mock.patch```.

You can read more about mock here:

http://www.voidspace.org.uk/python/mock/

**Comment by ankit01ojha** [comment]
Thanks @bowlofeggs and @trishnaguha , now it passed all the test .
screenshot from 2017-02-15 17-07-12

**Review by bowlofeggs — APPROVED** [review]
Nice, this seems to work!

**Comment by bowlofeggs** [comment]
There were three comments left that referenced the old location, so opted to just fix those in a separate PR: https://github.com/fedora-infra/bodhi/pull/1293

**Pull request merged**